### ISSUE TITLE
Test suite errors with error_highlight

### ISSUE THREAD

**Issue by voxik** [sinatra/sinatra#1774]
Running test suite against Ruby 3.1 without Bundler results in some errors:

~~~
$ ruby -e 'Dir.glob "./test/*_test.rb", &method(:require)'
cannot load such file -- asciidoctor: skipping asciidoc tests
cannot load such file -- builder: skipping builder tests
cannot load such file -- coffee-script: skipping coffee tests
cannot load such file -- creole: skipping creole tests
cannot load such file -- erubi
Did you mean?  erb: skipping erubi tests
cannot load such file -- erubis: skipping erubis tests
cannot load such file -- haml: skipping haml tests
cannot load such file -- less: skipping less tests
cannot load such file -- liquid: skipping liquid tests
cannot load such file -- markaby: skipping markaby tests
cannot load such file -- pandoc-ruby: skipping markdown tests with Tilt::PandocTemplate
cannot load such file -- commonmarker: skipping markdown tests with Tilt::CommonMarkerTemplate
cannot load such file -- redcarpet: skipping markdown tests with Tilt::RedcarpetTemplate
cannot load such file -- rdiscount: skipping markdown tests with Tilt::RDiscountTemplate
cannot load such file -- kramdown: skipping markdown tests with Tilt::KramdownTemplate
cannot load such file -- maruku: skipping markdown tests with Tilt::MarukuTemplate
cannot load such file -- bluecloth: skipping markdown tests with Tilt::BlueClothTemplate
cannot load such file -- wikicloth: skipping mediawiki tests
cannot load such file -- nokogiri: skipping nokogiri tests
cannot load such file -- rabl: skipping rabl tests
cannot load such file -- radius: skipping radius tests
cannot load such file -- sass: skipping sass tests
cannot load such file -- sass: skipping scss tests
cannot load such file -- slim: skipping slim tests
cannot load such file -- stylus: skipping stylus tests
cannot load such file -- redcloth: skipping textile tests
cannot load such file -- wlang: skipping wlang tests
cannot load such file -- yajl
Did you mean?  yaml: skipping yajl tests
Run options: --seed 33747

# Running:

........................................................................................................................................................................................................................................................................................................................................................F...................S...........................................................................................F....F..............................................................................................................................................................................................................................................F.................................................................................................................................................................................................................................

Finished in 2.596864s, 356.5840 runs/s, 811.3633 assertions/s.

  1) Failure:
SettingsTest::TestDump_errors#test_does_not_dump_404_errors_0 [/builddir/build/BUILD/sinatra-2.2.0/usr/share/gems/gems/sinatra-2.2.0/test/settings_test.rb:387]:
Expected false to be truthy.

  2) Failure:
RoutingTest#test_recalculates_body_length_correctly_for_404_response_0 [/builddir/build/BUILD/sinatra-2.2.0/usr/share/gems/gems/sinatra-2.2.0/test/routing_test.rb:174]:
--- expected
+++ actual
@@ -1 +1,3 @@
-"18"
+# encoding: US-ASCII
+#    valid: true
+"64"


  3) Failure:
RoutingTest#test_captures_the_exception_message_of_a_raised_NotFound_0 [/builddir/build/BUILD/sinatra-2.2.0/usr/share/gems/gems/sinatra-2.2.0/test/routing_test.rb:186]:
--- expected
+++ actual
@@ -1 +1,3 @@
-"19"
+# encoding: US-ASCII
+#    valid: true
+"99"


  4) Failure:
StaticTest#test_path_is_escaped_in_404_error_pages_0 [/builddir/build/BUILD/sinatra-2.2.0/usr/share/gems/gems/sinatra-2.2.0/test/static_test.rb:87]:
Unexpected response content..
--- expected
+++ actual
@@ -1 +1,4 @@
-["GET &#x2F;&lt;script&gt;"]
+["GET &#x2F;&lt;script&gt;
+
+        raise NotFound, &quot;\#{request.request_method} \#{request.path_info}&quot;
+        ^^^^^"]


926 runs, 2107 assertions, 4 failures, 0 errors, 1 skips

You have skipped tests. Run with --verbose for details.
~~~

These seems to be caused by error_highlight, because disabling it fixes the issues:

~~~
$ RUBYOPT=--disable-error_highlight ruby -e 'Dir.glob "./test/*_test.rb", &method(:require)'
cannot load such file -- asciidoctor: skipping asciidoc tests
cannot load such file -- builder: skipping builder tests
cannot load such file -- coffee-script: skipping coffee tests
cannot load such file -- creole: skipping creole tests
cannot load such file -- erubi
Did you mean?  erb: skipping erubi tests
cannot load such file -- erubis: skipping erubis tests
cannot load such file -- haml: skipping haml tests
cannot load such file -- less: skipping less tests
cannot load such file -- liquid: skipping liquid tests
cannot load such file -- markaby: skipping markaby tests
cannot load such file -- pandoc-ruby: skipping markdown tests with Tilt::PandocTemplate
cannot load such file -- commonmarker: skipping markdown tests with Tilt::CommonMarkerTemplate
cannot load such file -- redcarpet: skipping markdown tests with Tilt::RedcarpetTemplate
cannot load such file -- rdiscount: skipping markdown tests with Tilt::RDiscountTemplate
cannot load such file -- kramdown: skipping markdown tests with Tilt::KramdownTemplate
cannot load such file -- maruku: skipping markdown tests with Tilt::MarukuTemplate
cannot load such file -- bluecloth: skipping markdown tests with Tilt::BlueClothTemplate
cannot load such file -- wikicloth: skipping mediawiki tests
cannot load such file -- nokogiri: skipping nokogiri tests
cannot load such file -- rabl: skipping rabl tests
cannot load such file -- radius: skipping radius tests
cannot load such file -- sass: skipping sass tests
cannot load such file -- sass: skipping scss tests
cannot load such file -- slim: skipping slim tests
cannot load such file -- stylus: skipping stylus tests
cannot load such file -- redcloth: skipping textile tests
cannot load such file -- wlang: skipping wlang tests
cannot load such file -- yajl
Did you mean?  yaml: skipping yajl tests
Run options: --seed 28281

# Running:

.............................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................S................................................................................

Finished in 1.961836s, 472.0068 runs/s, 1075.5230 assertions/s.

926 runs, 2110 assertions, 0 failures, 0 errors, 1 skips

You have skipped tests. Run with --verbose for details.
~~~

**Comment by dentarg** [comment]
(Linking with https://github.com/sinatra/sinatra/issues/1772)

Are you running from the latest master branch?

This should be fixed by https://github.com/sinatra/sinatra/pull/1519 but I see that that was merged into `unstable` so maybe it is not in master? cc @jkowens

**Comment by jkowens** [comment]
@dentarg yes that is in master now 👍

**Comment by dentarg** [comment]
Can you link the commit @jkowens? I fail to spot it (although I'm on mobile right now browsing GitHub, so not using the best tool hehe)

**Comment by dentarg** [comment]
Okay, I see it now (using computer and a real shell :-) https://github.com/sinatra/sinatra/commit/bda8c29d

**Comment by jkowens** [comment]
Haha ok cool 😅

**Comment by dentarg** [comment]
@voxik Note that https://github.com/sinatra/sinatra/commit/bda8c29d has not been part of a release yet (despite its old age)

I can not spot it in https://github.com/sinatra/sinatra/tree/2-2-stable so it will probably only be released with Sinatra v3.0.0 (🤞 that that will happen), or what do you say @jkowens?

**Comment by jkowens** [comment]
@dentarg yeah I would love to see a 3.0 release sometime soon. I don't know when @namusyaka will have a chance to do that, but it's been a long time coming 😄

**Comment by voxik** [comment]
> (Linking with #1772)
> 
> Are you running from the latest master branch?

I was testing against 2.2.0 and the #1772 seems to be real life example of this test issue. BTW did_you_mean can exhibit similar behavior to error_highlight.

> This should be fixed by #1519 but I see that that was merged into `unstable` so maybe it is not in master? cc @jkowens

Thank you for pointing this patch out. It seems to fix this issue. Sorry for failing my search. It probably looked quite old.

**Comment by dentarg** [comment]
Sinatra 3.0.0 is out, I think we can close this issue now.
